marketo-client/client.go

```diff
@@ -0,0 +1,260 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package marketoclient
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/SpeakData/minimarketo"
+)
+
+var (
+	ErrEnqueueLimit = errors.New("enqueue limit reached")
+	ErrZeroRecords  = errors.New("no records found")
+	ErrCannotCancel = errors.New("cannot cancel export, since it is already in completed state")
+)
+
+type Client struct {
+	minimarketo.Client
+}
+
+func NewClient(config minimarketo.ClientConfig) (Client, error) {
+	client, err := minimarketo.NewClient(config)
+	if err != nil {
+		return Client{}, err
+	}
+	return Client{client}, nil
+}
+
+func (c Client) CreateExportLeads(fields []string, startDate string, endDate string) (string, error) {
```

Comment:
We should have a comment describing what the method does, what it takes as input and what it produces as output.

Reply:
added comments